### PR TITLE
Enabled calculation of grid spacing from lat/lon grid and spectral filtering of input data 

### PR DESCRIPTION
**Pull request by JuliaKukulies**
Made minor changes to `get_spacings()` in `tobac.utils.general` that allows for the calculation of horizontal grid spacing from lat/lon grid instead of only from cartesian grids. 

Also added a test for  `get_spacings()`  in  `test_utils.py` . This test file can be extended to also test other functions of the utils module.

### CODE REVIEW

**Comment by JuliaKukulies** [comment]
I also added a new functionality in `feature_detection.py,` where the input data can not only be filtered through a Gaussian filter, but also through a spectral 2D bandpass filter, where specific wavelengths can be filtered out. The user can specify a minimum and a maximum wavelength. This can be useful if specific meteorological scales are of interest or when cyclone tracking is done on rather noisy fields such as relative vorticity. Can remove this second feature again, if you think it is not needed.

**Review by freemansw1 — COMMENTED** [review]
Thanks so much for these changes! I'm really excited to get these in overall. I've left some comments in a few places; I think we need to resolve style/format before we can merge these in. Maybe we should put that on the agenda for the next development meeting (I'll e-mail you with the details).

File: tobac/themes/tobac_v1/feature_detection.py
Comment (JuliaKukulies) [reply tobac/themes/tobac_v1/feature_detection.py:187]: Agree, good point!

File: tobac/tests/test_utils.py
Comment (JuliaKukulies) [reply tobac/tests/test_utils.py:20]: I am actually not sure. Generating test data inside the function also makes sense, but with the parametrizing fixture `@pytest.mark.parametrize()` it is possible to test multiple datasets or sets of arguments in general. So if the test functions are going to be extended, this could be in advantage, because it can directly be tracked if the test fails for a specific combination of arguments.

File: tobac/themes/tobac_v1/feature_detection.py
Comment (fsenf) [reply tobac/themes/tobac_v1/feature_detection.py:611]: Yes, this would be required! We specified this aspect in our CONTRIBUTING.md file. Unfortunately, this file is a bit hidden in the v2.0-dev branch ....

Comment (JuliaKukulies) [reply tobac/themes/tobac_v1/feature_detection.py:611]: Ah perfect! I completely missed that file, but thank you, it is very clear.

File: tobac/themes/tobac_v1/feature_detection.py
Comment (freemansw1) [reply tobac/themes/tobac_v1/feature_detection.py:17]: Per our discussion at the meeting last week, I think the plan is to accept these changes and then eventually do a PR on `v2.0-dev` that has had Black run on it.

**Review by freemansw1 — APPROVED** [review]
I'm happy with these changes now! Great work!

**Comment by JuliaKukulies** [comment]
I made one more change, because I figured that it may be good for users to be able to check how the wavelength filter looks like for your specific data (it will become more clear in the example). But otherwise, I am happy to merge now if @fsenf and @w-k-jones approve. Many thanks for the help and review @freemansw1 ! 

Regarding the example notebook: Should I just submit a separate PR to https://github.com/climate-processes/tobac-tutorials? And is [this](https://zenodo.org/record/3195910#.YgvtR3VKj0o)  still the location to upload data? I use some WRF output files in the example that are not too large, so I could upload them there.

**Review by w-k-jones — APPROVED** [review]

**Comment by freemansw1** [comment]
Deferring to @fsenf on the tutorials question. I'm not sure that we've discussed the future of that repo yet? I'll leave this open for a few days so that we can hopefully get a resolution there; if not, I will merge early next week.

**Comment by fsenf** [comment]
Great to see progress! For tobac-tutorials:
* I started discussion on data upload here: https://github.com/climate-processes/tobac-tutorials/issues/8
* and I also prepared the update of the tutorials: https://github.com/climate-processes/tobac-tutorials/pull/7

@freemansw1 , @JuliaKukulies , @w-k-jones : Please have a look and contribute to the discussion! 

It would be great if your tutorial notebook (@JuliaKukulies ) already fits into the new tutorials structure.

**Comment by fsenf** [comment]
It looks good ... I let it go ;-)

**Pull request merged**
